index.js

```diff
@@ -161,7 +161,7 @@ function CompassCompiler(inputTree, files, options) {
   this.options = merge(true, this.defaultOptions);
   merge(this.options, options);
   options = this.options;
-  options.files = (options.files instanceof Array) ? options.files : [];
+  options.files = (options.files && options.files.length) ? [].concat.apply([], [options.files]) : [];
```

Comment:
Your solution also nice, but first you should validate existence of ```options.files.length``` to include only array and string. 

so it will be:
```js
options.files = options.files && options.files.length && [].concat(options.files) || [];
```